core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingServiceConfiguration.java

```diff
@@ -174,17 +176,24 @@ public BindingHandlerAdvise BindingHandlerAdvise(
 
 	@Bean
 	@ConditionalOnMissingBean(BinderFactory.class)
-	public BinderFactory binderFactory(BinderTypeRegistry binderTypeRegistry,
+	public DefaultBinderFactory binderFactory(BinderTypeRegistry binderTypeRegistry,
 			BindingServiceProperties bindingServiceProperties,
-			ObjectProvider<BinderCustomizer> binderCustomizerProvider) {
+			ObjectProvider<BinderCustomizer> binderCustomizerProvider,
+			BinderChildContextInitializer binderChildContextInitializer) {
 		DefaultBinderFactory binderFactory = new DefaultBinderFactory(
 				getBinderConfigurations(binderTypeRegistry, bindingServiceProperties),
 				binderTypeRegistry, binderCustomizerProvider.getIfUnique());
 		binderFactory.setDefaultBinder(bindingServiceProperties.getDefaultBinder());
 		binderFactory.setListeners(this.binderFactoryListeners);
+		binderChildContextInitializer.setBinderFactory(binderFactory);
```

Comment:
This avoid timing and circular dependencies between the 2. Really binder factory (BF) does not know about binderChildContextInitializer (BCCI) but we need to set the initializers early enough so that when FunctionConfiguration.afterPropertiesSet() starts requesting binders to bind the functions to that the AOT context is used and not the normal JVM one. 

> I will add the above to a shorter comment in the code as well